File: Code/Java-code/ServiceAppointment.java
public class ServiceAppointment{
    
    private String date;
    private String time;
    private BookingType booking;
    private PaymentRecievedType paid;
    private boolean destroy;
    private double cost;
    private CarStatusType status;

    private Car car;
    private Customer customer;
    
    public String getDate() {
        return date;
    }
    
    public void setDate(String date) {
        this.date = date;
    }
    
    public String getTime() {
        return time;
    }
    
    public void setTime(String time) {
        this.time = time;
    }
    
    public BookingType getBooking() {
        return booking;
    }
    
    public void setBooking(BookingType booking) {
        this.booking = booking;
    }
    
    public PaymentRecievedType getPaid() {
        return paid;
    }
    
    public void setPaid(PaymentRecievedType paid) {
        this.paid = paid;
    }
    
    public boolean isDestroy() {
        return destroy;
    }
    
    public void setDestroy(boolean destroy) {
        this.destroy = destroy;
    }
    
    public double getCost() {
        return cost;
    }
    
    public void setCost(double cost) {
        this.cost = cost;
    }
    
    public CarStatusType getStatus() {
        return status;
    }
    
    public void setStatus(CarStatusType status) {
        this.status = status;
    }

    public void cancelAppointment(){
    }

    
    public void setCar(Car car)
    {
        this.car = car;

    }

        
    public void setCustomer(Customer customer)
    {
        this.customer = customer;

    }

    public String toString(){
        return "Appointment date: " + date + " Booking type: " + booking + " with "+this.car +" for "+ this.customer;
    }
}

File: Code/PHP-Code/src/ServiceAppointment.php
<?php
require_once "autoload.php";
class ServiceAppointment{

    private string $date;
    private string $time;
    private BookingType $booking;
    private PaymentRecievedType $paid;
    private bool $destroy;
    private float $cost;

    private Car $car;
    private Customer $customer;

    public function getPaid(): PaymentRecievedType
    {
        return $this->paid;
    }

    public function setPaid(PaymentRecievedType $paid): void
    {
        $this->paid = $paid;
    }

    public function isDestroy(): bool
    {
        return $this->destroy;
    }

    public function setDestroy(bool $destroy): void
    {
        $this->destroy = $destroy;
    }

    public function getCost(): float
    {
        return $this->cost;
    }

    public function setCost(float $cost): void
    {
        $this->cost = $cost;
    }

    public function getStatus(): CarStatusType
    {
        return $this->status;
    }

    public function setStatus(CarStatusType $status): void
    {
        $this->status = $status;
    }
    private CarStatusType $status;

    public function getDate(): string
    {
        return $this->date;
    }

    public function setDate(string $date): void
    {
        $this->date = $date;
    }

    public function getTime(): string
    {
        return $this->time;
    }

    public function setTime(string $time): void
    {
        $this->time = $time;
    }

    public function getBooking(): BookingType
    {
        return $this->booking;
    }

    public function setBooking(BookingType $booking): void
    {
        $this->booking = $booking;
    }

    public function setCustomer(Customer $customer): void
    {
        $this->customer = $customer;
    }

    public function setCar(Car $car): void
    {
        $this->car = $car;
    }

    public function getCar(): Car
    {
        return $this->car;
    }

    public function __toString()
    {
        return "Appointment date: ". $this->date . " Appointment type: " . $this->booking->name ." with ". $this->car." for ". $this->customer->getName();
    }


}